extensions/vault/runtime/src/main/java/io/quarkus/vault/runtime/client/dto/sys/VaultUnwrapResult.java

```diff
@@ -0,0 +1,34 @@
+package io.quarkus.vault.runtime.client.dto.sys;
+
+import io.quarkus.vault.runtime.client.dto.AbstractVaultDTO;
+
+/**
+ * {
+ * "request_id": "5f654232-b0de-c27f-1fae-70c9d323c458",
+ * "lease_id": "",
+ * "lease_duration": 0,
+ * "renewable": false,
+ * "data": null,
+ * "warnings": null,
+ * "auth": {
+ * "client_token": "s.6tyZnRgG7TpDMPg135CEVMHk",
+ * "accessor": "Z00Qd1lU9apbt8eBIm1XUQHE",
+ * "policies": [
+ * "apps",
+ * "default"
+ * ],
+ * "token_policies": [
+ * "apps",
+ * "default"
+ * ],
+ * "identity_policies": null,
+ * "metadata": null,
+ * "orphan": false,
+ * "entity_id": "",
+ * "lease_duration": 2764800,
+ * "renewable": true
+ * }
+ * }
+ */
+public class VaultUnwrapResult extends AbstractVaultDTO<Object, VaultUnwrapAuth> {
```

Comment:
a `VaultUnwrapResult` cannot be made of a `VaultUnwrapAuth`. it has to be generic, because response could be wrapped.

Reply:
Ok I understand in general what you are mention here and I agree, what is not clear to me and I need to play is reproduce the example you provided before of `approle` with wrapped token and see all the outputs to see how are different between them and how we can implement it. I don't know when I will have time to do it.

About renames I agree with you to follow the Vault terminology, although my guess is that i terms of Docs, Vault documentation uses different concepts for the same, but let's use what you suggest here which is more clear.

Reply:
@lordofthejars +1, we've touched upon the same issue in the vault health PR, we don't need to match the low level Vault API property names at the Quarkus level, users will appreciate it.

Reply:
I am not sure if I will be able to work on this in the near time, I mean let's collaborate all of us in this.

Reply:
@vsevel do you think we could arrange a call (1 hour call) and do pair programming and fix everything?

Reply:
@lordofthejars I'm also quite stretched but will be happy to contribute with some comments :-). There is no rush, unless Vincent prefers it merged asap. Thanks

Reply:
Sure no worries, as I said to @vsevel if we can do a pair programming session together will be the best way to proceed as with one hour we can have everything done and reviewed at the same time.

Reply:
@lordofthejars I doubt we can get close to merge in 1 hour, considering even the early code needs some changes, and adding on top of that testing and documentation.
by experience it is going to take some effort. if you do not have time now, may be this is something that you can put aside for a while. I think there is no rush. I have given my input on the applicability, and no user has asked for it.
If you decided to pursue it, one thing I did to help you out is providing you with example responses fo the 2 main use cases: approle and client-token:

approle wrapping token creation:
```
/ # vault write -wrap-ttl=60s -f auth/approle/role/myapp/secret-id
Key                              Value
---                              -----
wrapping_token:                  s.QdjujN8PGGN8EICldXpFnSYO
wrapping_accessor:               NCk8gkK90xLNzFMcnTq68nY8
wrapping_token_ttl:              1m
wrapping_token_creation_time:    2020-03-31 16:29:05.5958584 +0000 UTC
wrapping_token_creation_path:    auth/approle/role/myapp/secret-id
```

approle unwrap response:
```
curl --header "X-Vault-Token: s.QdjujN8PGGN8EICldXpFnSYO" --request POST http://127.0.0.1:8200/v1/sys/wrapping/unwrap | jq
{
  "request_id": "5829832d-113a-48ed-7881-739d470fdbdf",
  "lease_id": "",
  "renewable": false,
  "lease_duration": 0,
  "data": {
    "secret_id": "ef8e1ebc-3be1-7bb7-3166-8b63b6289449",
    "secret_id_accessor": "e70b94da-e1ab-03c1-0fbc-8e73218f83b1"
  },
  "wrap_info": null,
  "warnings": null,
  "auth": null
}
```

client-token wrapping token creation:
```
/ # vault token create -wrap-ttl=60s -policy=myapp
Key                              Value
---                              -----
wrapping_token:                  s.xSr3IHx9RXxpSSwCctTd6n02
wrapping_accessor:               8H0BUXHPFv6LVIrXa9H9XNWb
wrapping_token_ttl:              1m
wrapping_token_creation_time:    2020-03-31 16:26:49.0405113 +0000 UTC
wrapping_token_creation_path:    auth/token/create
wrapped_accessor:                xoFYP9EgPi9I3xa7cDB3bm5v
```

client-token unwrap response:
```
curl --header "X-Vault-Token: s.xSr3IHx9RXxpSSwCctTd6n02" --request POST http://127.0.0.1:8200/v1/sys/wrapping/unwrap | jq
{
  "request_id": "7dd87b91-0c6b-c837-7918-4c377f7e635a",
  "lease_id": "",
  "renewable": false,
  "lease_duration": 0,
  "data": null,
  "wrap_info": null,
  "warnings": [
    "Policy \"myapp\" does not exist"
  ],
  "auth": {
    "client_token": "s.DiaodHOYFPxdsWbKtKX6qXr6",
    "accessor": "xoFYP9EgPi9I3xa7cDB3bm5v",
    "policies": [
      "default",
      "myapp"
    ],
    "token_policies": [
      "default",
      "myapp"
    ],
    "metadata": null,
    "lease_duration": 2764800,
    "renewable": true,
    "entity_id": "",
    "token_type": "service",
    "orphan": false
  }
}
```

so we can see that for the approle, the output is the original response, if it had not been wrapped. so in effect it is the response for the [Generate New Secret ID](https://www.vaultproject.io/api-docs/auth/approle/#sample-response-4)

since this operation has not been coded in `VaultClient` the data structure does not exist.
it is the same for the token creation.

however I do not think there is any challenge. just create any data structure you might need when unwrapping, then create a VaultClient `unwrap` operation that returns an `Object` with no unmarshalling (or just as jsonp objects), and depending on the use case unmarshall it on the right datastructure.

I do not think it is a huge piece of work, but it is definitely going to take some effort.